internal/controller/utils/nim.go

```diff
@@ -285,56 +271,80 @@ func requestToken(logger logr.Logger, req *http.Request) (*NimTokenResponse, err
 }
 
 // attemptToPullManifest is used for pulling a runtime for verifying access
-func attemptToPullManifest(logger logr.Logger, runtime NimRuntime, tokenResp *NimTokenResponse) error {
+func attemptToPullManifest(logger logr.Logger, runtime NimRuntime, tokenResp *NimTokenResponse) (*http.Response, error) {
 	req, reqErr := http.NewRequest("GET", fmt.Sprintf(nimGetRuntimeManifestFmt, runtime.Resource, runtime.Version), nil)
 	if reqErr != nil {
-		return reqErr
+		logger.Error(reqErr, "failed to construct request")
```

Comment:
imho, in this cases there is no need to log the error as it is being returned.

Reply:
Good catch!
Resolved in https://github.com/opendatahub-io/odh-model-controller/pull/390/commits/ee47edf344e97e497f6604ebf90fc802cf8147a9.